api/app/models/spree/api/webhooks/payment_decorator.rb

```diff
@@ -6,6 +6,17 @@ def after_void
           super
           queue_webhooks_requests!('payment.voided')
         end
+
+        def after_completed
+          super
+          order.queue_webhooks_requests!('order.paid') if all_valid_order_payments_completed?
+        end
+
+        private
+
+        def all_valid_order_payments_completed?
+          order.payments.valid.completed.size == order.payments.valid.size
```

Comment:
Let's use `Order#paid?` method here (plus update that method to not rely on payment_state and rely on this condition here + check if the valid payments amount cover order total)

Also, let's create a similar one called `Order#shipped?` for shipped event + add specs for these methods